### PR TITLE
allow header and set header in client

### PR DESCRIPTION
**Pull request by hunjixin**
need send Authorization header to remote exec cmds. 
allow header and support set header value in client

### CODE REVIEW

**Comment by hunjixin** [comment]
@Stebalien

**Comment by hunjixin** [comment]
anyone can review this pr?

File: http/config.go [hunk 1]
```diff
@@ -85,6 +85,12 @@ func (cfg *ServerConfig) SetAllowCredentials(flag bool) {
 	cfg.corsOpts.AllowCredentials = flag
 }
 
+func (cfg *ServerConfig) SetAllowHeaders(headers ...string) {
+	cfg.corsOptsRWMutex.Lock()
+	defer cfg.corsOptsRWMutex.Unlock()
+	cfg.corsOpts.AllowedHeaders = append(cfg.corsOpts.AllowedHeaders, headers...)
```
Comment (Stebalien) [inline http/config.go:91]: This should either be called `AddAllowedHeaders` or it should set (not append) the allowed headers.

File: http/client.go [hunk 2]
```diff
@@ -40,6 +41,16 @@ func ClientWithUserAgent(ua string) ClientOpt {
 	}
 }
 
+// ClientWithUserAgent specifies the HTTP user agent for the client.
```
Comment (Stebalien) [inline http/client.go:44]: Incorrect docs.

**Review by Stebalien — CHANGES_REQUESTED** [review]
The change seems reasonable, but please add a test in http/client_test.go (see the first test as a good example).

**Comment by BigLep** [comment]
@hunjixin: are you able to add the test so we can get this merged?

**Comment by hunjixin** [comment]
@BigLep  have tested

**Comment by BigLep** [comment]
Thanks.  We're blocked on finding maintainer review time.  It's in the queue though.

File: http/client.go [hunk 3]
```diff
@@ -40,6 +41,16 @@ func ClientWithUserAgent(ua string) ClientOpt {
 	}
 }
 
+// ClientWithHeaders specifies the HTTP header for the client.
+func ClientWithHeaders(key, value string) ClientOpt {
```
Comment (Stebalien) [inline http/client.go:45]: nit

```suggestion
// ClientWithHeader adds an HTTP header to the client.
func ClientWithHeader(key, value string) ClientOpt {
```

File: http/client_test.go [hunk 4]
```diff
@@ -91,3 +91,47 @@ func TestClientAPIPrefix(t *testing.T) {
...
+			if path := r.URL.Path; path != expPath {
+				t.Errorf("expected path %q, got %q", expPath, path)
+			}
+
+			w.WriteHeader(http.StatusOK)
+		}))
+		testClient := s.Client()
+		tc.host = s.URL
+		r := &cmds.Request{Path: tc.path, Command: &cmds.Command{}, Root: &cmds.Command{}}
+		c := NewClient(tc.host, ClientWithHeaders(tc.header, tc.value)).(*client)
```
Comment (Stebalien) [inline http/client_test.go:129]:
```suggestion
		c := NewClient(tc.host, ClientWithHeader(tc.header, tc.value)).(*client)
```

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
Merged in https://github.com/ipfs/go-ipfs-cmds/pull/226 (addressing my review). Thanks!

**Pull request merged**